mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -1384,6 +1397,12 @@ def log_batch(self, run_id, metrics, params, tags):
         _validate_batch_log_limits(metrics, params, tags)
         _validate_param_keys_unique(params)
 
+        is_single_metric = len(metrics) == 1
+        path = "metrics" if is_single_metric else ""
+        for index, metric in enumerate(metrics):
+            _validate_metric(metric.key, metric.value, metric.timestamp, metric.step, path=path)
+            path = path if is_single_metric else append_to_json_path(path, f"[{index}]")
```

Comment:
The fact that `_validate_metric` needs `path` is strange here, since it's very specific to the filestore. Can we create a separate method and call it, instead of constructing a bunch of dummy paths?

Reply:
`path` here means the json path of the current value (e.g, `metrics[0].value`), not a file path. It's confusing tho.